### PR TITLE
feat(hogql): properties between shards

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Accessing an event's person's properties in HogQL would fetch the entire properties object and send it back to us the main select query.

## Changes

Extracts properties within the person subquery and only returns those. Cutting down a lot on traffic between shards. 

I do not have benchmarks, but looking at the queries, the effects should be similar to this: https://github.com/PostHog/posthog/pull/13935

## How did you test this code?

Wrote tests. Checked that all worked in the browser.

### CODE REVIEW

File: posthog/api/test/__snapshots__/test_query.ambr [hunk 1]
```diff
@@ -237,9 +237,38 @@
   /* user_id:0 request:_snapshot_ */
   SELECT event,
          events.distinct_id,
-         replaceRegexpAll(JSONExtractRaw(events.properties, 'key'), '^"|"$', ''),
+         replaceRegexpAll(JSONExtractRaw(properties, 'key'), '^"|"$', ''),
```
Comment (mariusandra) [inline posthog/api/test/__snapshots__/test_query.ambr:240]: Previously `events.` was added before `properties` to disambiguate it from the `events__pdi__person.properties` column that we selected in the join. That's no longer the case and thus it seems like it was automatically removed.

File: posthog/hogql/test/test_printer.py [hunk 2]
```diff
@@ -60,16 +60,29 @@ def test_fields_and_properties(self):
...
+
+        context = HogQLContext(within_non_hogql_query=False, using_person_on_events=False)
+        self.assertEqual(
+            self._expr("person.properties.bla", context),
+            "events__pdi__person.properties___bla",
+        )
+
+        context = HogQLContext(within_non_hogql_query=False, using_person_on_events=True)
+        self.assertEqual(
+            # TODO: for now, explicitly writing "poe." to opt in. Automatic switching will come soon.
```
Comment (mariusandra) [inline posthog/hogql/test/test_printer.py:83]: HogQL is not yet able to turn PoE (in its current form) on or off depending on the team's settings. It is doing this inside existing insights (non-hogql queries), but not in full hogql queries. This is planned for another PR.

**Review by mariusandra — COMMENTED** [review]
Ready for a look

**Review by Twixes — APPROVED** [review]
Looking really good!

File: posthog/models/event/events_query.py [hunk 3]
```diff
@@ -22,13 +21,28 @@
...
+    "properties",
+    "timestamp",
+    "team_id",
+    "distinct_id",
+    "elements_chain",
+    "created_at",
+    "person_id",
+    "person.created_at",
+    "person.properties.name",
+    "person.properties.email",
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply posthog/models/event/events_query.py:37]: Mainly for speed. I don't think we need to return the entire properties object with every event. There are rumours of possible customers incoming with 10k+ properties per person, etc. This makes it more in line with what selecting "person" returns in this table.

**Pull request merged**